### DISCUSSION TITLE
How to Set Timer using Elsa.Activities.Temporal nuget package in ELSA 3.1.0 to create timer based workflow

### DISCUSSION THREAD

**Discussion by NiveditaGalwade** [elsa-workflows/elsa-core#5311] [Q&A]
How to invoke workflow based on timer which is present in Elsa.Activities.Temporal nuget package in ELSA 3.1.0 
below is sample workflow. Please suggest correct answer considering  ELSA 3.1.0
 public class TimerWorkflow(IServiceCollection serviceCollection) : IWorkflow
 {
     public void Build(IWorkflowBuilder builder)
     {

         builder
             
        .Timer() // Getting error here to set timer
        .WriteLine(context => $"Timer event at {DateTime.Now}");
     }
 }

**Comment by glime-ai** [comment]
To invoke a workflow based on a timer in Elsa 3.1.0, you need to ensure that you have the `Elsa.Activities.Temporal` NuGet package installed, as it contains the timer activities. Once you have that package, you can define a timer in your workflow using the `Timer` activity.

Here's an example of how you might define a workflow with a timer:

```csharp
using Elsa.Activities.Temporal;
using Elsa.Builders;

public class TimerWorkflow : IWorkflow
{
    public void Build(IWorkflowBuilder builder)
    {
        builder
            .Timer(Duration.FromSeconds(30)) // Set the timer to trigger every 30 seconds
            .WriteLine(context => $"Timer event at {DateTime.Now}");
    }
}
```

In the above example, `Duration.FromSeconds(30)` specifies that the timer should trigger every 30 seconds. You can adjust the duration according to your needs.

Please note that in Elsa 3.x, the way to specify the timer interval has changed from previous versions. You now use the `Duration` class to specify the interval.

Make sure that you have the correct using directives at the top of your file to reference the necessary namespaces for the timer and workflow builder classes.

If you are still encountering an error, please provide the exact error message for further assistance.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
